src/bin/vip-import-sql.js

```diff
@@ -35,13 +37,35 @@ const appQuery = `
 	environments{ id, appId, type, name, syncProgress { status }, primaryDomain { name } }
 `;
 
-const err = message => {
-	console.log( chalk.red( message.toString().replace( /^(Error: )*/, 'Error: ' ) ) );
-	process.exit( 1 );
-};
-
 const debug = debugLib( 'vip:vip-import-sql' );
 
+const gates = async ( app, env, fileName ) => {
+	const { id: envId, appId } = env;
+	const track = trackEventWithEnv.bind( null, appId, envId );
+
+	if ( ! currentUserCanImportForApp( app ) ) {
+		exit.withError(
+			'The currently authenticated account does not have permission to perform a SQL import.'
+		);
+	}
+
+	if ( ! isSupportedApp( app ) ) {
+		await track( 'import_sql_command_error', { error_type: 'unsupported-app' } );
+		exit.withError( 'The type of application you specified does not currently support SQL imports.' );
+	}
+
+	const fileSize = await getFileSize( fileName );
+
+	if ( ! fileSize ) {
+		exit.withError( `File '${ fileName }' is empty.` );
+	}
+
+	if ( fileSize > SQL_IMPORT_FILE_SIZE_LIMIT ) {
+		exit.withError(
+			`The sql import file size (${ fileSize } bytes) exceeds the limit the limit (${ SQL_IMPORT_FILE_SIZE_LIMIT } bytes).` +
+				'Please split it into multiple files or contact support for assistance.' );
```

Comment:
> Please split it into multiple files

Is this an opportunity to provide a link to the documentation on how you can do it? Or we can just leave "Contact support for assistance" now.

Reply:
TBH I'm open to either.  This code was moved in this PR but not fabricated.  Happy to adjust as necessary.